### DISCUSSION TITLE
CircleNode Radius and Map Projection Problem

### DISCUSSION THREAD

**Discussion by mehmetalideligoz** [pelicanmapping/osgearth#2377] [Support Q&A]
Hello,
I want to show a CircleNode in the map. But the radius that I see is not right when I actually measure it (google earth).

```
  osg::ref_ptr<CircleNode> circle = new CircleNode();
  circle->setPosition(center); // center has the same srs as the map
  circle->setRadius(Distance(2000, Units::METERS));
```

When I create a line by using GeoMath::destination, I can compute the correct point.

```
auto centerPoint = m_Center.transform(m_Center.getSRS()->getGeographicSRS());
    double outLon;
    double outLat;
    GeoMath::destination(osgEarth::deg2rad(centerPoint.y()),
      osgEarth::deg2rad(centerPoint.x()), osgEarth::deg2rad(angle),
      radius.as(Units::METERS), outLat, outLon); // move 2km

    centerPoint.x() = osgEarth::rad2deg(outLon);
    centerPoint.y() = osgEarth::rad2deg(outLat);
    const auto endPoint = wgsCenterPoint.transform(m_Center.getSRS());
```

![image](https://github.com/gwaldron/osgearth/assets/14887939/a2147a7f-2a0b-4e49-b9d5-a5be6ef61586)

The map profile and the ImageLayer profile are the same (the reasoın was GDAL::reproject takes too much time, I've decied to equalize my XYZImage profile and the map profile).
> Profile: +proj=webmerc +a=6378137 +b=6378137 +lon_0=0.0 +x_0=0.0 +y_0=0 +k=1.0 +units=m +nadgrids=@null +towgs84=0,0,0,0,0,0,0 +no_defs

The map source is http://[abc].tile.openstreetmap.org/{z}/{x}/{y}.png (spherical-mercator)

osgearth version is 3.5.0

## Note
If I do not equalize the map profile and image layer profile the resulting circle radius is correct. 
![image](https://github.com/gwaldron/osgearth/assets/14887939/bf49bbb1-b6f8-4439-a750-cb1817dd0e2c)

What am I missing?
Thanks

## Update
Maybe I should use FeatureNode for that? When I manually create a feature and compute the geometry using GeoMath::destination it looks correct. Sorry if it is a dumb question :)

**Comment by gwaldron** [comment]
Help me understand better. Can you post a simple earth file with your set up?
